connector/saml/saml.go

```diff
@@ -348,19 +353,97 @@ func (p *provider) HandlePOST(s connector.Scopes, samlResponse string) (ident co
 	return ident, nil
 }
 
+func (p *provider) validateStatus(resp *response) error {
+	status := resp.Status
+	if status == nil {
+		return fmt.Errorf("response did not contain a Status")
+	}
+	statusCode := status.StatusCode
+	if statusCode == nil {
+		return fmt.Errorf("response did not contain a StatusCode")
+	}
+	if statusCode.Value != "urn:oasis:names:tc:SAML:2.0:status:Success" {
+		return fmt.Errorf("status code of the response was not Success")
+	}
+	return nil
+}
+
+func (p *provider) validateSubjectConfirmation(subject *subject) error {
+	subjectConfirmation := subject.SubjectConfirmation
+	if subjectConfirmation == nil {
+		return fmt.Errorf("response did not contain a SubjectConfirmation")
+	}
+	subjectConfirmationData := subjectConfirmation.SubjectConfirmationData
+	if subjectConfirmationData == nil {
+		return fmt.Errorf("response did not contain a SubjectConfirmationData")
+	}
+	if subjectConfirmationData.Recipient != "" && subjectConfirmationData.Recipient != p.redirectURI {
+		return fmt.Errorf("expected recipient %q got %q", p.redirectURI, subjectConfirmationData.Recipient)
+	}
+	return nil
+}
+
+func (p *provider) validateConditions(assertion *assertion) error {
+	conditions := assertion.Conditions
+	if conditions == nil {
+		return fmt.Errorf("response did not contain a Conditions element")
+	}
+	now := p.now()
+	notBefore := time.Time(conditions.NotBefore)
```

Comment:
You are right. The timestamps in the conditions element are optional and should only be validated if present.
I will change the code accroding to https://github.com/onelogin/python-saml/blob/master/src/onelogin/saml2/response.py#L317-L335